src/shogun/base/SGObject.h

```diff
@@ -1065,15 +1087,70 @@ class SGObject: public std::enable_shared_from_this<SGObject>
 	 * @param _tag name information of parameter
 	 * @param value new value of parameter
 	 */
-	void update_parameter(const BaseTag& _tag, const Any& value, bool do_checks = true);
+	template <typename T>
+	void update_parameter(const BaseTag& _tag, const T& value, bool do_checks = true)
```

Comment:
made this template dependent so that we can have all these parameter handling branches in a single location